pkg/azure/actuator.go

```diff
@@ -131,54 +121,43 @@ func (a *Actuator) needsUpdate(ctx context.Context, cr *minterv1.CredentialsRequ
 	if !exists {
 		return true, nil
 	}
+	// passthrough-specifc checks here (now the only kind of checks...)
 
-	azureStatus, err := decodeProviderStatus(a.codec, cr)
+	credentialsRootSecret, err := a.GetCredentialsRootSecret(ctx, cr)
 	if err != nil {
-		return true, fmt.Errorf("unable to decode ProviderStatus: %v", err)
+		log.WithError(err).Debug("error retrieving cloud credentials secret")
+		return false, err
+	}
+	// If the cloud credentials secret has been updated in passthrough mode, we need an update
+	if credentialsRootSecret != nil && credentialsRootSecret.ResourceVersion != cr.Status.LastSyncCloudCredsSecretResourceVersion {
+		logger.Debug("root cloud creds have changed, update is needed")
+		return true, nil
 	}
 
-	cloudCredsSecret, err := a.GetCredentialsRootSecret(ctx, cr)
-	if err != nil {
-		logger.WithError(err).Error("issue with cloud credentials secret")
+	// If the target Secret data doesn't match the cloud credentials secret (we haven't yet pivoted to passthrough from mint)
+	// then we need an update
+	targetSecretKey := types.NamespacedName{Namespace: cr.Spec.SecretRef.Namespace, Name: cr.Spec.SecretRef.Name}
+	targetSecret := &corev1.Secret{}
+	if err := a.client.Get(ctx, targetSecretKey, targetSecret); err != nil {
+		logger.WithError(err).Error("failed to fetch target secret")
 		return true, err
 	}
 
-	azureCredentialsMinter, err := a.credentialMinterBuilder(
-		logger,
-		string(cloudCredsSecret.Data[AzureClientID]),
-		string(cloudCredsSecret.Data[AzureClientSecret]),
-		string(cloudCredsSecret.Data[AzureTenantID]),
-		string(cloudCredsSecret.Data[AzureSubscriptionID]),
-	)
-	if err != nil {
-		return true, fmt.Errorf("unable to create azure cred minter: %v", err)
+	if string(targetSecret.Data[AzureClientID]) != string(credentialsRootSecret.Data[AzureClientID]) ||
+		string(targetSecret.Data[AzureClientSecret]) != string(credentialsRootSecret.Data[AzureClientSecret]) {
+		return true, nil
 	}
```

Comment:
It really shouldn't change as the cluster being pivoted from mint to passthrough will still be running under the same subscriptionID and tenantID. In fact, I might even argue that including subscription and tenant information as part of the azure secret format was an unfortunate mistake. That information should have been kept (and referenced when needed) in the Infrastructure CR since it is shared cluster-wide information.